sleap_roots/bases.py

```diff
@@ -69,36 +69,44 @@ def get_base_tip_dist(
     return distances
 
 
-def get_base_xs(base_pts: np.ndarray) -> np.ndarray:
+def get_base_xs(base_pts: np.ndarray) -> np.ndarray | np.floating:
     """Get x coordinates of the base of each lateral root.
 
     Args:
         base_pts: root bases as array of shape `(instances, 2)` or `(2)` when there is
             only one root, as is the case for primary roots.
 
     Return:
-        An array of base x-coordinates (instances,) or (1,) when there is only one root.
+        An array of base x-coordinates (instances,) or a scalar when there is only one root.
     """
-    if base_pts.ndim not in (1, 2):
-        raise ValueError(
-            "Input array must be 2-dimensional (instances, 2) or 1-dimensional (2,)."
-        )
-    if base_pts.shape[-1] != 2:
-        raise ValueError("Last dimension must be (x, y).")
-
-    base_xs = base_pts[..., 0]
+    # Check for the 2D shape of the input array
+    if base_pts.ndim == 1:
+        # If shape is `(2,)`, then reshape it to `(1, 2)` for consistency
+        base_pts = base_pts.reshape(1, 2)
+    elif base_pts.ndim != 2:
```

Comment:
do make sure to add tests for new conditionals to the appropriate modules (test_bases).